src/entries/popup/pages/home/TokenDetails/TokenContextMenu.tsx

```diff
@@ -76,17 +96,54 @@ export function TokenContextMenu({ children, token }: TokenContextMenuProps) {
     navigate(ROUTES.BRIDGE);
   };
 
-  const isBridgeable = token.bridging?.isBridgeable;
+  const togglePinToken = useCallback(() => {
+    if (pinned) {
+      removedPinnedAsset({ uniqueId: token.uniqueId });
+      triggerToast({
+        title: i18n.t('token_details.toast.unpin_token', {
+          name: token.symbol,
+        }),
+      });
+      simulateClick(containerRef.current);
+      return;
+    }
+    addPinnedAsset({ uniqueId: token.uniqueId });
+    triggerToast({
+      title: i18n.t('token_details.toast.pin_token', {
+        name: token.symbol,
+      }),
+    });
+    simulateClick(containerRef.current);
+  }, [containerRef, token, pinned, addPinnedAsset, removedPinnedAsset]);
 
-  const explorer = getTokenBlockExplorer(token);
-  const isNative = isNativeAsset(token?.address, token?.chainId);
+  const hideToken = useCallback(() => {
+    addHiddenAsset({ uniqueId: computeUniqueIdForHiddenAsset(token) });
+    if (pinned) removedPinnedAsset({ uniqueId: token.uniqueId });
+    setSelectedToken();
+    triggerToast({
+      title: i18n.t('token_details.toast.hide_token', {
+        name: token.symbol,
+      }),
+    });
+  }, [token, pinned, addHiddenAsset, removedPinnedAsset, setSelectedToken]);
+
+  const copyTokenAddress = useCallback(() => {
+    if (isNative) return;
+    copyAddress(token.address);
+    simulateClick(containerRef.current);
+  }, [token.address, containerRef, isNative]);
 
   if (isWatchingWallet && !allowSwap && isNative) return <>{children}</>;
 
   return (
     <DetailsMenuWrapper closed={true} onOpenChange={onOpenChange}>
       <ContextMenuTrigger asChild>{children}</ContextMenuTrigger>
       <ContextMenuContent>
+        <TokenMenuShortcutListener
+          hideToken={hideToken}
+          pinToken={togglePinToken}
+          copyTokenAddress={copyTokenAddress}
+        />
```

Comment:
Do you think this functionality could live within `useTokenShortcuts` with the other token shortcut handlers?

Reply:
Yeah was thinking of doing that too, but realised we already have it as a global scope [here](https://github.com/rainbow-me/browser-extension/blob/master/src/entries/popup/pages/home/Tokens.tsx#L229) in the homepage. I could take the hook and listen to it again, but we would face duplications which would cause double tracking.